lib/pcre.mli

```diff
@@ -29,6 +29,12 @@ val exec : rex:regexp -> ?pos:int -> string -> groups
 val get_substring : groups -> int -> string
 (** Equivalent to {!Core.Group.get}. *)
 
+val names : regexp -> string array
```

Comment:
List types are usually more straight forward for plug in into idiomatic OCaml APIs.

Reply:
I know, I even spontaneously put a list here in my first implementation! But I switched to an array to keep the same API as pcre.

Reply:
Note that this a compatibility API. If you are willing to make more changes in your code, it is probably better to directly use Re.group_names, which is more useful and efficient.